lib/_http_incoming.js

```diff
@@ -137,130 +137,101 @@ function _addHeaderLines(headers, n) {
-function matchKnownFields(field) {
-  var low = false;
-  while (true) {
-    switch (field) {
-      case 'Content-Type':
-      case 'content-type':
-        return 'content-type';
-      case 'Content-Length':
-      case 'content-length':
-        return 'content-length';
-      case 'User-Agent':
-      case 'user-agent':
-        return 'user-agent';
-      case 'Referer':
-      case 'referer':
-        return 'referer';
-      case 'Host':
-      case 'host':
-        return 'host';
-      case 'Authorization':
-      case 'authorization':
-        return 'authorization';
-      case 'Proxy-Authorization':
-      case 'proxy-authorization':
-        return 'proxy-authorization';
-      case 'If-Modified-Since':
-      case 'if-modified-since':
-        return 'if-modified-since';
-      case 'If-Unmodified-Since':
-      case 'if-unmodified-since':
-        return 'if-unmodified-since';
-      case 'From':
-      case 'from':
-        return 'from';
-      case 'Location':
-      case 'location':
+function matchKnownFields(field, lowercased) {
+  switch (field.length) {
+    case 3:
+      if (field === 'Age' || field === 'age') return 'age';
+      break;
+    case 4:
+      if (field === 'Host' || field === 'host') return 'host';
+      if (field === 'From' || field === 'from') return 'from';
+      if (field === 'ETag' || field === 'etag') return 'etag';
+      if (field === 'Date' || field === 'date') return '\u0000date';
+      if (field === 'Vary' || field === 'vary') return '\u0000vary';
+      break;
+    case 6:
+      if (field === 'Server' || field === 'server') return 'server';
+      if (field === 'Cookie' || field === 'cookie') return '\u0002cookie';
+      if (field === 'Origin' || field === 'origin') return '\u0000origin';
+      if (field === 'Expect' || field === 'expect') return '\u0000expect';
+      if (field === 'Accept' || field === 'accept') return '\u0000accept';
+      break;
+    case 7:
+      if (field === 'Referer' || field === 'referer') return 'referer';
+      if (field === 'Expires' || field === 'expires') return 'expires';
+      if (field === 'Upgrade' || field === 'upgrade') return '\u0000upgrade';
+      break;
+    case 8:
+      if (field === 'Location' || field === 'location')
         return 'location';
-      case 'Max-Forwards':
-      case 'max-forwards':
-        return 'max-forwards';
-      case 'Retry-After':
-      case 'retry-after':
-        return 'retry-after';
-      case 'ETag':
-      case 'etag':
-        return 'etag';
-      case 'Last-Modified':
-      case 'last-modified':
-        return 'last-modified';
-      case 'Server':
-      case 'server':
-        return 'server';
-      case 'Age':
-      case 'age':
-        return 'age';
-      case 'Expires':
-      case 'expires':
-        return 'expires';
-      case 'Set-Cookie':
-      case 'set-cookie':
+      if (field === 'If-Match' || field === 'if-Match')
```

Comment:
Isn't this supposed to be `if-match` instead of `if-Match`?

Reply:
Yes, I'll change it.